code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm

```diff
@@ -245,46 +256,354 @@
 		return
 
-	if(victim.stat == DEAD)
+	var/mob/living/carbon/xenomorph/hunter/X = owner
+
+	if(!isliving(A))
+		if(!silent)
+			to_chat(X, "<span class='xenowarning'>We cannot sting this target!</span>")
+		return FALSE
+
+	var/mob/living/sting_target = A
+
+	if(!A?.can_sting())
+		if(!silent)
+			to_chat(X, "<span class='xenowarning'>We cannot sting this target!</span>")
+		return FALSE
+
+	if(X.on_fire)
+		if(!silent)
+			to_chat(X, "<span class='xenowarning'>We're too busy being on fire to sting them!</span>")
+		return FALSE
+
+	if(!X.Adjacent(sting_target))
+		if(!silent)
+			to_chat(X, "<span class='xenowarning'>We must be adjacent to the target.</span>")
+		return FALSE
+
+
+	if(sting_target.stat == DEAD)
+		if(!silent)
+			to_chat(X, "<span class='xenowarning'>We care not for the deceased!</span>")
+		return FALSE
+
+	if(ishuman(sting_target))
+		var/mob/living/carbon/human/human_target = A
+
+		if(isnestedhost(human_target)) //no bully
+			if(!silent)
+				to_chat(X, "<span class='xenowarning'>We refrain from unnecessarily bullying the host.</span>")
+			return FALSE
+
+	if(!SEND_SIGNAL(X, COMSIG_HUNTER_SNEAK_ATTACK_CHECK) & COMSIG_HUNTER_SNEAK_ATTACK)
+		if(!silent)
+			to_chat(X, "<span class='xenowarning'>We must be able to sneak attack the target to properly sting it!</span>")
+		return FALSE
+
+	var/datum/action/xeno_action/stealth/sneak_attack_check = X.actions_by_path[/datum/action/xeno_action/stealth]
+	if(sneak_attack_check.last_run_intent > (world.time - HUNTER_SNEAK_ATTACK_RUN_DELAY))//We can't sprint up to the target and use this.
+		if(!silent)
+			to_chat(X, "<span class='xenowarning'>We must be stalking or stationary to properly sting the target!</span>")
+		return FALSE
+
+	return TRUE
+
+
+/datum/action/xeno_action/activable/sneak_stinger/on_cooldown_finish()
+	to_chat(owner, "<span class='xenowarning'><b>Our toxin glands refill, allowing us to sting our victims.</b></span>")
+	owner.playsound_local(owner, 'sound/voice/alien_drool1.ogg', 25, 0, 1)
+	return ..()
+
+
+/datum/action/xeno_action/activable/sneak_stinger/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/X = owner
+	var/mob/living/victim = A
+
+	X.face_atom(victim)
+
+	to_chat(X, "<span class='xenodanger'>We prepare to sting our quarry...</span>")
+
+	if(!do_after(X, HUNTER_STINGER_WINDUP, TRUE, target, BUSY_ICON_HOSTILE)) //Slight wind up
+		to_chat(X, "<span class='xenodanger'>We aborting stinging [victim].</span>")
+		return fail_activate()
+
+	victim.attack_alien(X) //We auto-sneak attack as part of this ability.
+	var/inject_toxin = /datum/reagent/toxin/acid/xeno_acid
+	var/transfer_amount = HUNTER_SNEAK_ATTACK_INJECT_AMOUNT
+	if(X.a_intent != INTENT_HARM) //Inject neurotoxin instead of acid while on non-harm intent and double the dose
+		inject_toxin = /datum/reagent/toxin/xeno_neurotoxin
+		transfer_amount *= 2
+		if(X.a_intent == INTENT_HELP) //So we do *something* visual
+			X.do_attack_animation(victim)
+
+	var/atom/reagent_name = inject_toxin
```

Comment:
This is not an atom

Reply:
This is just used to get the name of the reagent.

Reply:
This is not an atom
It is a datum